src/RazorSdk/SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -181,21 +193,26 @@ private static IReadOnlyList<TagHelperDescriptor> ResolveTagHelperDescriptors(Ge
             return result;
         }
 
-        private static IReadOnlyList<TagHelperDescriptor> GetTagHelperDescriptorsFromReferences(Compilation compilation, StaticCompilationTagHelperFeature tagHelperFeature)
+        private static IReadOnlyList<TagHelperDescriptor> GetTagHelperDescriptorsFromReferences(GeneratorExecutionContext context, StaticCompilationTagHelperFeature tagHelperFeature)
         {
             List<TagHelperDescriptor> tagHelperDescriptors = new();
+            var compilation = context.Compilation;
 
             foreach (var reference in compilation.References)
             {
                 var guid = reference.GetModuleVersionId(compilation);
-                IReadOnlyList<TagHelperDescriptor> descriptors = default;
+                IReadOnlyList<TagHelperDescriptor> descriptors = new List<TagHelperDescriptor>();
 
                 if (guid is Guid _guid)
                 {
                     if (!_tagHelperCache.TryGetValue(_guid, out descriptors))
                     {
                         tagHelperFeature.TargetReference = reference;
                         descriptors = tagHelperFeature.GetDescriptors();
+                        context.ReportDiagnostic(Diagnostic.Create(
```

Comment:
Could we include the number of descriptors found?

Reply:
Sure. Done.